keps/sig-network/0015-20180614-SCTP-support.md

```diff
@@ -297,11 +298,71 @@ We describe in the Kubernetes documentation the mutually exclusive nature of use
 
 ### Risks and Mitigations
 
+## Test Plan
+
+Unlike with UDP and TCP, we can't necessarily just test that SCTP
+connections work, because some machines won't have the SCTP kernel
+module installed (or will have it blocked from being loaded). So we
+want to have some tests that are just "make sure kube-proxy is doing
+what we expect it to" that can run everywhere, and then another set of
+"SCTP connections actually work" tests that are behind a feature tag.
+
+For the standard PR/periodic tests, we'll add:
+
+- Something like `"Pods, Services, and NetworkPolicies can declare
+  SCTP ports"`.
+
+  - If using the `iptables` proxier, this will additionally check that
+    appropriate iptables rules are created.
+
+  - The test also ensures that if `sctp.ko` was not loaded before the
+    test (on the node with the created test pod), then it is still not
+    loaded after the test. (FIXME: that could fail if it ran in
+    parallel with some of the other tests below.)
+
+  - (FIXME: how does e2e work with feature gates? This test and the
```

Comment:
I actually don't know the answer to this any more.  @BenTheElder can you comment on testing patterns for stuff like this, or refer to someone who knows best?

Reply:
it's not super great, we often have a seperate copy of some standard job with the feature gate togggled in the deployer and then including the `[Feature:Foo]` ginkgo test tags.

toggling it will be dependent on the test setup.

imo we could do with a more robust strategy for this, i'm not really sure we have one.